src/averageAges.js

```diff
@@ -15,11 +15,14 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const rightCentury = century
+    ? people.filter(person => Math.ceil(person.died / 100) === century)
+    : people;
+  const menAges = rightCentury
+    .filter(person => person.sex === 'm')
+    .map(person => person.died - person.born);
+
+  return menAges.reduce((age, sum) => sum + age, 0) / menAges.length;
```

Comment:
Same here, instead of using map and reduce - you can just calculate age in reduce

Reply:
And same logic will be used in next function, so will be better to create helper function for it